core/src/main/java/org/hibernate/ogm/service/impl/DefaultSchemaInitializationContext.java

```diff
@@ -84,6 +86,19 @@ public Database getDatabase() {
 		return allIdSourceKeyMetadata;
 	}
 
+	@Override
+	public Map<String, Class<?>> getTableEntityTypeMapping() {
+		Map<String, Class<?>> mapping = new HashMap<>();
+		Map<String, EntityPersister> entityPersisters = factory.getEntityPersisters();
+
+		for ( Entry<String, EntityPersister> entityPersisterEntry : entityPersisters.entrySet() ) {
```

Comment:
A table might contain several entities (inheritance). Wondering whether that may be a problem here?